### PR TITLE
Implement config, update search options

### LINKED ISSUE #4: Implement Basic Search API
Our Search API should provide a polymorphic list of index cards, support a paginated scroll. The default query should be based on current date-time and the user-roles of the person making the request. A plain text search for any card with the provided text is MVP. 

NOTE: Research how OpenSearch Kibana might be used to support this service

### LINKED ISSUE #13: Add jest unit tests
checked = added
- [ ] config.ts
- [ ] app.ts
- [ ] client.ts
- [ ] querybody.ts
- [ ] errorHandler.ts

### LINKED ISSUE #15: Add values to config of the API
Currently, the API Config() lacks setup for generating links to related services in the search UI. We need to enhance the config object to incorporate environment variables for specifying the hostname and port for different types of objects stored in the search database. This configuration is crucial for enabling the search UI to seamlessly link to other services.

When users perform a search, the search UI displays a list of index cards representing objects matching the search criteria. Each index card contains one or more links to edit the corresponding object. However, since the routes to these edit pages may reside on different ports, simple relative links are insufficient.

The current idea is to configure the hostname and port number for each service in DNS and pass them to the API via environment variables. This configuration will hopefully enable the API to dynamically generate correct links to related services based on the search results.

Any suggestions or ideas regarding the implementation are welcome and encouraged.

### PR DESCRIPTION
**Pull request by FlatBallFlyer**
This refactor implements a seperation of concerns, jest unit testing, and stepci end-to-end testing.

# Seperation of concerns
/src/config - Configuration Code
/src/controllers - HTTP request/reply handlers
/src/services - Service layer for RBAC and Business Logic
/src/store - ElastiicIO wrapper
/src/server.ts - Entrypoint to Initilize and Run Server

closes #15 
closes #4 
closes #13 

 NOTE: These changes are dependent on code from PR's on the mentorhub and mentorhub-elasticsearch repo's. To test locally you will need to
- Clone mentorhub, set branch to kafka-cdc and run ``make install`` to get docker-compose updates
- Clone mentorhub-elasticsearch, set branch to version-1.0.0 and run ``npm run container`` - new initialize
You should then be able to run ``npm run container`` and ``npm run stepci`` to see the API in action.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,22 @@
 {
+  "name": "mentorhub-search-api",
+  "description": "This is the API used to support search use cases in the mentorhub system",
   "scripts": {
-    "build": "webpack",
-    "start": "nodemon src/server.ts",
-    "test": "jest"
+    "build": "tsc",
+    "prestart": "mh down, mh up elasticsearch",
```
Comment (michquinn) [inline package.json:6]: Syntax? (I would think the comma should be a semicolon, no?)

File: src/config/Config.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+export  class Config {
+    private static instance: Config; // Singleton 
+
+    configItems: ConfigItem[] = [];
+    apiVersion: string = "";
+
+    // Private Properties
+    #configFolder: string = "./";
+    #port: number = 8084;
```
Comment (michquinn) [inline src/config/Config.ts:25]: Why 8084?

**Review by michquinn — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
+FROM node:21 AS build
+
+#Install git (for branch/patch info)
+WORKDIR /app
+COPY package*.json .
```
[the remark this answers is not in the document]
Comment (FlatBallFlyer) [reply Dockerfile:5]: corrected

File: package.json (hunk 1 above)
Comment (FlatBallFlyer) [reply package.json:6]: corrected - not sure why I didn't test this

File: src/config/Config.ts (hunk 2 above)
Comment (FlatBallFlyer) [reply src/config/Config.ts:25]: See system port number specifications here: https://github.com/agile-learning-institute/mentorHub/blob/main/specifications/architecture.yaml. Basically all the API's are eventually served from the same host, just at different port numbers.

Comment (michquinn) [reply src/config/Config.ts:25]: I'm still unclear as to why it's 8084 here and 8081 elsewhere in the file

Comment (FlatBallFlyer) [reply src/config/Config.ts:25]: you are correct, will fix this

Comment (FlatBallFlyer) [reply src/config/Config.ts:25]: OH - that is an un-needed initialization, it is over-ridden as the default on line 43 - I chaned this to 0

**Review by michquinn — APPROVED** [review]

**Pull request merged**